javascript/ql/src/Security/CWE-367/FileSystemRace.qhelp

```diff
@@ -0,0 +1,61 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+
+<overview>
+<p>
+Often it is necessary to check the state of a file before using it. These checks usually take a file name
+to be checked, and if the check returns positively, then the file is opened or otherwise operated upon.
+</p>
+
+<p>
+However, in the time between the check and the operation, the underlying file referenced by the
+file name could be changed by an attacker, causing unexpected behavior.
+</p>
+
+</overview>
+
+<recommendation>
+<p>
+Use file descriptors instead of file names whenever possible.
+</p>
+</recommendation>
+
+<example>
+<p>
+The following example shows a case where it is checked whether a file inside the <code>/tmp/</code> folder exists, 
```

Comment:
To avoid passive voice
```suggestion
The following example shows a case where the code checks whether a file inside the <code>/tmp/</code> folder exists, 
```